vault/resource_kv_secret_v2.go

```diff
@@ -108,6 +147,33 @@ func getKVV2Path(mount, name, prefix string) string {
 	return fmt.Sprintf("%s/%s/%s", mount, prefix, name)
 }
 
+func getKVV2MetadataPath(mount, name string) string {
+	return fmt.Sprintf("%s/metadata/%s", mount, name)
+}
+
+func getCustomMetadata(d *schema.ResourceData) map[string]interface{} {
+	data := map[string]interface{}{}
+
+	metadataFields := []string{
+		consts.FieldMaxVersions,
+		consts.FieldCASRequired,
+		consts.FieldDeleteVersionAfter,
+		consts.FieldData,
+	}
+	fieldPrefix := fmt.Sprintf("%s.0", consts.FieldCustomMetadata)
+	for _, k := range metadataFields {
+		fieldKey := fmt.Sprintf("%s.%s", fieldPrefix, k)
+		vaultStateKey := k
```

Comment:
nit: you might be able to drop this intermediate variable in favor of assigning directly to `k` on line `168` below.